### PR TITLE
change lock to interlocked atomics

### PR DESCRIPTION
**Pull request by MadL1me**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Using `lock` to increment `_txCounter`

Issue Number: N/A

## What is the new behavior?

Now implementation uses lock-free CAS based value computation, which would work faster, without allocating new object and creating a monitor.

## Other information

None

### CODE REVIEW

**Review by rekby — APPROVED** [review]

**Comment by rekby** [comment]
Hello, sorry for late response.

Thanks for the fix :)
Can you rebase your PR and fix [linter issues](https://github.com/ydb-platform/ydb-dotnet-sdk/actions/runs/7842000595/job/21399441339?pr=59)?

File: src/Ydb.Sdk/src/Services/Table/Transaction.cs [hunk 1]
```diff
@@ -13,6 +13,8 @@ internal Transaction(string txId)
     public string TxId { get; }
 
     internal int? TxNum { get; private set; }
```
Comment (asmyasnikov) [inline src/Ydb.Sdk/src/Services/Table/Transaction.cs:15]:
```suggestion
    internal float? TxNum { get; private set; }
```

File: src/Ydb.Sdk/src/Services/Table/Transaction.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@ internal Transaction(string txId)
...
+
```
Comment (asmyasnikov) [inline src/Ydb.Sdk/src/Services/Table/Transaction.cs:16]:
```suggestion

```

File: src/Ydb.Sdk/src/Services/Table/Transaction.cs [hunk 3]
```diff
@@ -23,26 +24,17 @@ internal Transaction(string txId)
 
         var tx = new Transaction(
             txId: proto.Id);
+
```
Comment (asmyasnikov) [inline src/Ydb.Sdk/src/Services/Table/Transaction.cs:27]:
```suggestion

```

File: src/Ydb.Sdk/src/Services/Table/Transaction.cs [hunk 4]
```diff
@@ -25,23 +26,16 @@ internal Transaction(string txId)
...
         }
 
         return tx;
     }
-
-    private static readonly object TxCounterLock = new();
-    private static int _txCounter;
-
-    private static int GetTxCounter()
+
```
Comment (rekby) [inline src/Ydb.Sdk/src/Services/Table/Transaction.cs:35]:
```suggestion

```

**Pull request closed without merging**